src/api-umbrella/utils/interval_lock.lua

```diff
@@ -0,0 +1,78 @@
+local lock = require "resty.lock"
+
+local _M = {}
+
+--- Only one thread can execute fn through its execution duration
+-- A lock (mutex) is held while the function executes and released at the end
+-- @param name - a unique identifier for this lock (automatically namespaced)
+-- @param fn - function to execute if the lock can be held
+-- @return ok, error
+_M.mutex_exec = function(name, fn)
+  local check_lock = lock:new("locks", {["timeout"] = 0})
+  local _, lock_err = check_lock:lock("mutex_exec:" .. name)
+  if lock_err then
+    return true, nil
+  else
+    local results = pcall(fn)
```

Comment:
Logging the `pcall` results makes a lot of sense; adding. Thinking through the calls further, it doesn't seem like the results could be used meaningfully (as they're tied to a timer, outside of the calling thread), so I'll go ahead and remove them.